db/migrate/20170810164131_rename_provisioning_keys_to_environments.rb

```diff
@@ -0,0 +1,22 @@
+class RenameProvisioningKeysToEnvironments < ActiveRecord::Migration[5.0]
+  def change
+    rename_table :provisioning_keys, :environments
+    rename_column :environments, :api_env, :name
+    rename_column :environments, :content, :provisioning_key
+
+    add_reference :access_requests, :environment
+    add_reference :tokens,          :environment
+
+    Token.all.each do |token|
+      env = Environment.find_by(name: token.api_env)
+      token.environment_id = env.id
+      token.save!
+    end
+
+    AccessRequest.all.each do |access_request|
+      env = Environment.find_by(name: access_request.api_env)
+      access_request.environment_id = env.id
+      access_request.save!
+    end
```

Comment:
This should probably be written as a:
```
def up
  ...
end

def down
  ...
end
```
In order to make it reversible. So the `up` method would apply the changes that migrate the data and remove the `api_env` from `AccessRequest` and `Token` and the `down` method would do the reverse of that - restore the `api_env` field and move the string values back into it. `change` method is good for simple migrations where Rails can figure out what the reverse/rollback should be.